google/cloud/bigtable/tests/instance_admin_async_future_integration_test.cc

```diff
@@ -177,21 +187,37 @@ TEST_F(InstanceAdminAsyncFutureIntegrationTest,
   google::cloud::bigtable::CompletionQueue cq;
   std::thread pool([&cq] { cq.Run(); });
 
-  // create cluster
-  auto clusters_before = instance_admin_->ListClusters(id);
-  ASSERT_FALSE(IsClusterPresent(clusters_before, cluster_id_str))
+  // Make an asynchronous request, but immediately block because this is just a
+  // test.
+  auto clusters_list_before = instance_admin_->AsyncListClusters(cq, id).get();
+  EXPECT_TRUE(clusters_list_before.failed_locations.empty())
+      << "The Cloud Bigtable service (or emulator) reports that it could not"
+      << " retrieve the information for some locations. This is typically due"
+      << " to an outage or some other transient condition.";
+  ASSERT_FALSE(IsClusterPresent(clusters_list_before.clusters, cluster_id_str))
```

Comment:
IsClusterPresent takes a cluster name, not a cluster id.  The original code was wrong.  Please use `cluster.name()` instead of id.

Reply:
At that line (197) the cluster variable has not been created yet.
It is created at line 206.
So I added these  extra-checks at line 219 and 220:
  EXPECT_FALSE(IsClusterPresent(clusters_list_before.clusters, cluster.name()));
  EXPECT_TRUE(IsClusterPresent(clusters_list_after.clusters, cluster.name()));